gdraw/ggdkdraw.c

```diff
@@ -1573,7 +1612,13 @@ static void GGDKDrawSetTransientFor(GWindow transient, GWindow owner) {
     }
 
     if (owner == (GWindow) - 1) {
-        ow = gdisp->last_nontransient_window;
+        for (GList_Glib *pw = gdisp->toplevel_windows->head; pw != NULL; pw = pw->next) {
+            GGDKWindow tw = (GGDKWindow)pw->data;
+            if (tw != gw && tw->is_visible) {
+                ow = tw;
+                break;
+            }
+        }
```

Comment:
So to confirm, the idea hear is to find *some* window for this one to be transient to *now*, regardless of whether there's an actual arrangement and even if that window may be destroyed before this one, because GDK needs it for `gdk_window_set_transient_for`? And we won't wind up with a "dangling transient" because of the loop in `_GGDKDraw_OnWindowDestroyed` (which calls back into here)?

Reply:
The intent is to select the most recently focused window, and it shouldn't matter what gets destroyed first, because of the transient reparenting code. 

In terms of there being an 'actual arrangement' - there's no way at this point to know what the arrangement should be (and is one of my gripes with GDraw - it should have made everything explicit about what it's transient to), so selecting the most recently focused window is a best-effort attempt at that.

This is basically what the `last_nontransient_window` was trying to do, but it also works in the case where the `last_nontransient_window` is being destroyed, which causes `last_nontransient_window` to be unset (and before it gets set again) - this is the bulk of the real fix.

To be honest though, I wouldn't expect the transient reparenting code to ever run, but in the case it does happen, we can't destroy the transient window, because it's more than likely the caller will call GDrawDestroyWindow on it (and you can't destroy more than twice).

Reply:
Ah, OK, so the theory is that the most recently focused window is the best candidate for a transient parent for a new window, and the fact that it's also the target of the reparenting code (because it calls the same function) is more or less irrelevant, because that code is expected to run rarely or never.

Reply:
Yep, that's my current theory